mailgreen/controller/subscription_controller.py

```
@@ -0,0 +1,52 @@
+class SenderCount(BaseModel):
+    sender: str
+    name: str
+    count: int
+
+
+@router.post("/sync", response_model=List[SubscriptionOut])
+async def sync_subscriptions(user_id: UUID, db: Session = Depends(get_db)):
+    try:
+        return sync_subscriptions(db, str(user_id))
+    except Exception as e:
+        raise HTTPException(status_code=500, detail=str(e))
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)